### PR TITLE
Redesign error decorator, so it fits needs of both API versions (#374 part5)

### PR DESCRIPTION
**Pull request by inikulin**
\cc @kirovboris @churkin 

@kirovboris note that `errorDecorator` has been changed. I've bumped the version already, so once we've done with this PR you will be able to update it for the studio.

### CODE REVIEW

**Comment by kirovboris** [comment]
lgtm

**Comment by inikulin** [comment]
ping @churkin

File: test/server/legacy-test-run-error-formatting-test.js [hunk 1]
```diff
@@ -13,60 +13,14 @@ var untestedErrorTypes = Object.keys(TYPE).map(function (key) {
...
-            },
-
-            'strong': function (str) {
-                return str;
-            },
-
-            'a': function (str) {
-                return '"' + str + '"';
-            }
+        decorator ['span category'] = function (str) {
```
Comment (churkin) [inline test/server/legacy-test-run-error-formatting-test.js]:
```javascript
decorator ['span category'] = str => 'CATEGORY=' + str + '\n';
```

**Comment by churkin** [comment]
\r-

File: test/server/legacy-test-run-error-formatting-test.js (hunk 1 above)
Comment (inikulin) [inline test/server/legacy-test-run-error-formatting-test.js]: It's a test. We can't use ES6 in tests

**Comment by churkin** [comment]
lgtm

**Pull request merged**